### PR TITLE
dispatch a pause event before hiding window on darwin

### LINKED ISSUE #52: video keeps playing on window close
### steps to repro

On OS X:

screen shot 2016-03-06 at 11 31 59 am

Play a video. Hit the little red button. Sound keeps going.

Since I think we're just showing/hiding the window on close the video never gets paused.

### PR DESCRIPTION
**Pull request by Flet**
Fixes #52

### CODE REVIEW

File: main/windows.js [hunk 1]
```diff
@@ -48,7 +48,9 @@ function createMainWindow (menu) {
   win.on('close', function (e) {
     if (process.platform === 'darwin' && !isQuitting) {
       e.preventDefault()
-      win.hide()
+      win.webContents.executeJavaScript('dispatch("pause")', function () {
+        win.hide()
+      })
```
Comment (feross) [inline main/windows.js]: We should find a cleaner way to dispatch this. I'll merge then fix.

File: main/windows.js [hunk 2]
```diff
@@ -48,7 +48,9 @@ function createMainWindow (menu) {
   win.on('close', function (e) {
     if (process.platform === 'darwin' && !isQuitting) {
       e.preventDefault()
-      win.hide()
+      win.webContents.executeJavaScript('dispatch("pause")', function () {
```
Comment (feross) [inline main/windows.js]: It'd be better to use IPC for this. I'll merge and fix up.

File: main/windows.js (hunk 1 above)
Comment (feross) [inline main/windows.js]: Done here: https://github.com/feross/webtorrent-app/commit/a6e8a3462ee1241b9fb5e24fb023cc43060b6b95

**Comment by Flet** [comment]
Thanks ipc looks a lot cleaner!

**Pull request merged**